data-prepper-core/src/main/java/com/amazon/dataprepper/parser/PipelineParser.java

```diff
@@ -99,6 +101,19 @@ private void buildPipelineFromConfiguration(
             final List<List<Processor>> processorSets = pipelineConfiguration.getProcessorPluginSettings().stream()
                     .map(this::newProcessor)
                     .collect(Collectors.toList());
+
+            final List<List<Processor>> decoratedProcessorSets = processorSets.stream()
```

Comment:
I'm not sure that this is quite the best place to do this. Right now, it is simple because you the `PeerForwardingProcessorDecorator` takes in nothing other than the `Processor`. But, it will become more complicated later when it has to take in the PeerForwardingClient and PeerForwardingServer.

Here are a couple of options.
1. Supply a more general `ProcessorFunction` which can can perform transformations on processors.
2. Add another class which calls `PipelineParser` and then transforms the processor.

I think I like option 1, so I'll add a skeleton of how it could work.


Add this into the constructor.
```
public PipelineParser(final String pipelineConfigurationFileLocation, final PluginFactory pluginFactory, final List<Function<Processor, Processor>> processorTransformers) {
    // ...
    this.processorTransformers = processorTransformers;
}
```

In `buildPipelineFromConfiguration`:

```
processorSets.stream()
  .map(processorSet -> processorSet.stream()
    .map(processor -> processorTransformers.map(transformer -> transformer.apply(processor))
    .collect...
```
You might want to double check that mapping. But the idea is that each processor transformer has an opportunity to modify it.

Then you can implement the conditional logic as one of the processor transformers.

Reply:
This change is going beyond the scope of this PR. I'll try to create another PR for this.

Reply:
That works for me!